doc/source/user_guide/fields/field_data.rst

```diff
@@ -147,98 +161,34 @@ First, create a transaction object for field data.
   >>> transaction = solver.fields.field_data.new_transaction()
   # This creates a new transaction object for batching multiple requests.
 
-Then combine requests for multiple fields using ``add_<items>_request`` methods in a single transaction:
-
-- ``add_surfaces_request`` adds a surfaces request.
-- ``add_scalar_fields_request`` adds a scalar fields request.
-- ``add_vector_fields_request`` adds a vector fields request.
-- ``add_pathlines_fields_request`` adds a pathlines fields request.
+Next, combine requests for multiple fields using the ``add_requests`` method within a single transaction,
+then call the ``get_response`` method to execute the transaction and retrieve the data.
 
 .. code-block:: python
 
-  >>> transaction.add_surfaces_request(
-  >>>     surfaces=[1], data_types=[SurfaceDataType.Vertices, SurfaceDataType.FacesCentroid]
-  >>> )
-  # Adds a request for surface data such as vertices and centroids.
-  >>> transaction.add_scalar_fields_request(
-  >>>     surfaces=[1, 2], field_name="pressure", node_value=True, boundary_value=True
-  >>> )
-  # Adds a request for scalar field data like pressure.
-  >>> transaction.add_vector_fields_request(
-  >>>     surfaces=[1, 2], field_name="velocity"
-  >>> )
-  # Adds a request for vector field data like velocity.
+  >>> surface_request_1 = SurfaceFieldDataRequest(surfaces=[1], data_types=[SurfaceDataType.Vertices, SurfaceDataType.FacesCentroid])
+  >>> scalar_request_1 = ScalarFieldDataRequest(surfaces=[1, 2], field_name="pressure", node_value=True, boundary_value=True)
+  >>> vector_request_1 = VectorFieldDataRequest(surfaces=[1, 2], field_name="velocity")
 
-You can call the ``get_fields`` method to execute the transaction and retrieve the data:
+  >>> payload_data = transaction.add_requests(surface_request_1, scalar_request_1, vector_request_1).get_response()
 
-.. code-block:: python
-
-  >>> payload_data = transaction.get_fields()
-  # Executes all requests and returns the combined field data.
+You can use the ``get_field_data`` method on ``payload_data`` to retrieve data in the same way
+as fetching field data. You may either reuse the existing request object or create a new one
```

Comment:
@prmukherj "in the same way as fetching field data" is sketchy. The user is defintely confused by this statement.